tests/testthat/test-unnecessary_placeholder_linter.R

```diff
@@ -1,8 +1,9 @@
+linter <- unnecessary_placeholder_linter()
```

Comment:
Defining `linter` at the top level lets it leak out of the test file, potentially causing confusing error messages down the line if another test forgets to define the linter.

I think we should refrain from that.

Reply:
I don't think that's so -- my understanding is each file is run in its own environment.

I tested this like so:

```r
writeLines(con  = "test-a.R", '
a = 1

test_that("a = 1", {
  expect_equal(a, 1)
})')

writeLines(con = "test-b.R", '
stopifnot(!exists("a"))

b = 3
test_that("b = 3", {
  expect_equal(b, 3)
})')

testthat::test_dir(".")
✔ | F W S  OK | Context
✔ |         1 | a [0.1s]                                                                                                                                                                                                                                                                  
✔ |         1 | b                                                                                                                                                                                                                                                                         

══ Results ═══════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════
Duration: 0.2 s

[ FAIL 0 | WARN 0 | SKIP 0 | PASS 2 ]
```

Reply:
How are we including `helper.R` to make it survive? 😅

Reply:
That's done here AIUI

https://github.com/r-lib/testthat/blob/4c6bf287f59d5e48470d267f5792cd0eeb4af319/R/test-files.R#L210

```r
writeLines("c = 1", "helper.R")

writeLines(con  = "test-a.R", '
stopifnot(exists("c"))
a = 1

test_that("a = 1", {
  expect_equal(a, 1)
})
')

writeLines(con = "test-b.R", '
stopifnot(exists("c"))
stopifnot(!exists("a"))

b = 3
test_that("b = 3", {
  expect_equal(b, 3)
})
')

testthat::test_dir(".")
# ✔ | F W S  OK | Context
# ✔ |         1 | a [0.2s]                                                                                                                                                                                                    
# ✔ |         1 | b                                                                                                                                                                                                           

# ══ Results ═════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════
# Duration: 0.3 s

# [ FAIL 0 | WARN 0 | SKIP 0 | PASS 2 ]
```

Reply:
Ah, interesting. Thanks for the enlightenment.

Reply:
Related -- I had asked if {withr} would support a `local*` function to make this clearer:

https://github.com/r-lib/withr/issues/205

Seems they recommend `rlang::local_bindings()` to do that instead -- happy to make that change if you think it'll make it clearer rather than relying on this testthat behavior which is not well-known / particularly well-documented.

(will merge for now)